### DISCUSSION TITLE
the cost of travel

### DISCUSSION THREAD

**Discussion by GanHY97** [google/or-tools#4206] [Routing (and legacy CP) questions]
I want to consider the issue of different time periods and driving times when defining the cost of travel, but I don't know how to handle it.
For example, the travel time from A to B is 30 minutes from 8:00 to 12:00, and 40 minutes from 12:00 to 18:00
May I ask how to implement this

**Comment by jmarca** [comment]
This has been asked many times. Do a search. I’ve asked this myself when I first started with IR Tools. The short answer is you can’t, and you probably don’t really need to. The long answer is you’ll have to figure a work around somehow if you really need it, like dummy nodes, one per vehicle, that represent entering a congested time period and/or zone. Many other strategies have been discussed in the past. Do a search of github and the forum. JamesOn Apr 27, 2024, at 09:34, GanHY97 ***@***.***> wrote:﻿
I want to consider the issue of different time periods and driving times when defining the cost of travel, but I don't know how to handle it.
For example, the travel time from A to B is 30 minutes from 8:00 to 12:00, and 40 minutes from 12:00 to 18:00
May I ask how to implement this

**Reply by GanHY97** [reply]
Thank you very much, I will carefully search for the previous discussion again.

**Comment by kishan8354** [comment]
The cost of travel to Jaipur can vary widely based on your preferences and choices. Domestic flights to Jaipur range from ₹2,000 to ₹10,000, while train fares are more economical, between ₹500 and ₹2,000. Once in Jaipur, local transportation like auto-rickshaws and taxis offer convenient travel, with fares typically around ₹10-₹20 per kilometer. Accommodations range from budget hotels and hostels costing ₹500 to ₹1,500 per night to luxury hotels priced between ₹7,000 and ₹20,000 per night. Dining options are equally diverse, with street food costing as low as ₹50 and meals at mid-range restaurants ranging from ₹500 to ₹1,500 for two. Sightseeing costs include entrance fees of ₹50 to ₹500 for major attractions like the [Amber Fort](https://www.viewvoyager.in/tourist-destinations-in-jaipur/) and City Palace, with guided tours adding another ₹1,000 to ₹3,000. Budget travelers can comfortably explore Jaipur for ₹1,500 to ₹2,500 per day, while a more luxurious experience might cost upwards of ₹10,000 per day.
